tools/Google.Cloud.Tools.VersionCompat/Detectors/Class.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Tools.VersionCompat.CecilUtils;
+using Google.Cloud.Tools.VersionCompat.Utils;
+using Mono.Cecil;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Google.Cloud.Tools.VersionCompat.Detectors
+{
+    internal class Class
+    {
+        public static IEnumerable<Diff> Diffs(TypeDefinition o, TypeDefinition n)
+        {
+            var cls = new Class(o, n);
+            var classAndInterface = new ClassAndInterface(o, n);
+            return Enumerable.Empty<Diff>()
+                .Concat(cls.SealedAbstractStatic())
+                .Concat(cls.BaseType())
+                .Concat(classAndInterface.ImplementedInterfaces())
+                .Concat(classAndInterface.GenericConstraints())
+                .Concat(cls.Methods())
+                .Concat(cls.Properties());
+        }
+
+        private Class(TypeDefinition o, TypeDefinition n) => (_o, _n) = (o, n);
+
+        private readonly TypeDefinition _o;
```

Comment:
Rename to more useful names? I assume this is "original" and "new"?

Reply:
I'd rather keep the names very short as they are used a lot throughout the code.
Comment added to briefly explain what they mean.